.github/workflows/periodic.yml

```diff
@@ -0,0 +1,46 @@
+name: Periodic Functional Tests
+
+on:
+  schedule:
+    - cron: '*/5 * * * *' # Run every 5 minutes
+
+jobs:
+  run_tests:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      jobs:
+      setup_kind:
+      runs-on: ubuntu-latest
+      steps:
+    - name: Check if Docker is installed
+      run: |
+        if ! command -v docker &> /dev/null
+        then
+            echo "Docker is not installed. Installing Docker..."
+            sudo apt-get update
+            sudo apt-get install -y docker.io
+        else
+            echo "Docker is already installed."
+        fi
+    - name: Install KinD
```

Comment:
GH actions have an extension to provision KinD. Check this example: https://github.com/kubeflow/pipelines/blob/master/.github/workflows/backend.yml#L36